staging/src/k8s.io/apimachinery/pkg/runtime/converter.go

```diff
@@ -340,15 +472,22 @@ func sliceFromUnstructured(sv, dv reflect.Value) error {
 		return nil
 	}
 	dv.Set(reflect.MakeSlice(dt, sv.Len(), sv.Cap()))
+
+	pathLen := len(ctx.parentPath)
+	defer func() {
+		ctx.parentPath = ctx.parentPath[:pathLen]
+	}()
 	for i := 0; i < sv.Len(); i++ {
-		if err := fromUnstructured(sv.Index(i), dv.Index(i)); err != nil {
+		origLen := ctx.pushIndex(i)
```

Comment:
slight preference for dropping return values from pushIndex and pushKey, using the captured `pathLen` at the start of the method, and simply doing `ctx.parentPath = ctx.parentPath[:pathLen]` (applies everywhere)

it's easier to see by inspection what is happening without diving into two different methods